pandas/core/groupby/ops.py

```diff
@@ -375,7 +379,9 @@ def _reconstruct_ea_result(self, values, res_values):
         """
         # TODO: allow EAs to override this logic
 
-        if isinstance(values.dtype, (BooleanDtype, _IntegerDtype, FloatingDtype)):
+        if isinstance(
```

Comment:
it would be better to have a propertyfunction on an EA to determine what to do here (basically the comment above).

this is ok for backport but @debnathshoham can you create an issue about this.

Reply:
can you add the issue number you created in the TODO above